sql/2024/accessibility/a11y_overall_tech_usage_by_domain_rank.sql

```diff
@@ -0,0 +1,52 @@
+#standardSQL
+# Overall Accessibility (A11y) technology, ie. Overlays, usage by domain rank
+
+# Main SELECT statement to aggregate results by client and rank grouping.
+SELECT
+  client,
+  rank_grouping, # Grouping of domains by their rank (e.g., top 1000, top 10000, etc.)
+  total_in_rank, # Total number of sites within the rank grouping
+  COUNT(DISTINCT page) AS sites_with_a11y_tech, # Number of unique sites that use accessibility technology
+  COUNT(DISTINCT page) / total_in_rank AS pct_sites_with_a11y_tech # Percentage of sites using accessibility technology within the rank grouping
+FROM
+  (
+    # Subquery to filter and extract relevant pages with A11Y technology
+    SELECT DISTINCT
+      client,
+      page,
+      rank_grouping,
+      category
+    FROM
+      `httparchive.all.pages`,
+      UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping, # Expand rank_grouping to cover different rank categories
+      UNNEST(technologies) AS tech,
+      UNNEST(categories) AS category
+    WHERE
+      date = '2024-06-01' AND
+      category = 'Accessibility' AND
+      rank <= rank_grouping # Include only sites within the specified rank grouping
+  )
+JOIN
+  (
+    # Subquery to count total sites in each rank grouping for each client
+    SELECT
+      client,
+      rank_grouping,
+      COUNT(0) AS total_in_rank
+    FROM
+      `httparchive.all.pages`,
+      UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping
+    WHERE
+      date = '2024-06-01' AND
+      rank <= rank_grouping
+    GROUP BY
+      client,
+      rank_grouping
```

Comment:
Oh one thing is the `all` dataset has the home pages (is_root_page) and one secondary page per origin. I forgot to take that into account in my example I converted for you.

You should decide if you want to ONLY run this for the home page like in previous years (in which case add `AND is_root_page`) or if it would be interesting to see both (in which case should add `is_root_page` column so you can filter and pivot on those in the sheets).

Alternatively you can look at the combo of both but report at site level (e.g. a site uses a feature if it's in either the home page or the secondary page). But that one's probably a bit trickier as requires ensuring you count total `root_page`s rather than total `pages` so need to check that.

Reply:
I do like the idea of seeing both, and actually seeing an average of the two.   
  
I'll have to look into this later and maybe chat with the folks in the Accessibility channel / GitHub issues too.   
  
A secondary page and may not be as well monitored as the primary page.  Even knowing the spread between on average could be interesting. Home pages are often more complicated too (which tends to reduce their accessibility).

Reply:
@tunetheweb I am a bit confused as to why this isn't working. Isn't this where I should be adding the is\_root\_page to be able to have the extra column of data? 

```
#standardSQL
# Overall Accessibility (A11y) technology, ie. Overlays, usage by domain rank


# Main SELECT statement to aggregate results by client and rank grouping.
SELECT
client,
rank_grouping, # Grouping of domains by their rank (e.g., top 1000, top 10000, etc.)
total_in_rank, # Total number of sites within the rank grouping
COUNT(DISTINCT page) AS sites_with_a11y_tech, # Number of unique sites that use accessibility technology
COUNT(DISTINCT page) / total_in_rank AS pct_sites_with_a11y_tech # Percentage of sites using accessibility technology within the rank grouping
FROM
(
# Subquery to filter and extract relevant pages with A11Y technology
SELECT DISTINCT
client,
page,
rank_grouping,
category,
is_root_page
FROM
`httparchive.sample_data.pages_1k`,
UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping, # Expand rank_grouping to cover different rank categories
UNNEST(technologies) AS tech,
UNNEST(categories) AS category
WHERE
date = '2024-06-01' AND
category = 'Accessibility' AND
rank <= rank_grouping # Include only sites within the specified rank grouping
)
JOIN
(
# Subquery to count total sites in each rank grouping for each client
SELECT
client,
rank_grouping,
COUNT(0) AS total_in_rank
FROM
`httparchive.sample_data.pages_1k`,
UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping
WHERE
date = '2024-06-01' AND
rank <= rank_grouping
GROUP BY
client,
rank_grouping
)
USING (client, rank_grouping)
GROUP BY
rank_grouping,
total_in_rank,
client
ORDER BY
client,
rank_grouping
```

Reply:
Looks like you missed adding `is_root_page` from a few places. This seems to work:

```sql
#standardSQL
  # Overall Accessibility (A11y) technology, ie. Overlays, usage by domain rank
  # Main SELECT statement to aggregate results by client and rank grouping.
SELECT
  client,
  is_root_page,
  rank_grouping,
  # Grouping of domains by their rank (e.g., top 1000, top 10000, etc.)
  total_in_rank,
  # Total number of sites within the rank grouping
  COUNT(DISTINCT page) AS sites_with_a11y_tech,
  # Number of unique sites that use accessibility technology
  COUNT(DISTINCT page) / total_in_rank AS pct_sites_with_a11y_tech # Percentage of sites using accessibility technology within the rank grouping
FROM (
    # Subquery to filter and extract relevant pages with A11Y technology
  SELECT
    DISTINCT client,
    is_root_page,
    page,
    rank_grouping,
    category
  FROM
    `httparchive.sample_data.pages_1k`,
    UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping,
    # Expand rank_grouping to cover different rank categories
    UNNEST(technologies) AS tech,
    UNNEST(categories) AS category
  WHERE
    -- date = '2024-06-01' AND
    category = 'Accessibility' AND
    rank <= rank_grouping # Include only sites within the specified rank grouping
    )
JOIN (
    # Subquery to count total sites in each rank grouping for each client
  SELECT
    client,
    rank_grouping,
    COUNT(0) AS total_in_rank
  FROM
    `httparchive.sample_data.pages_1k`,
    UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping
  WHERE
    --date = '2024-06-01' AND
    rank <= rank_grouping
  GROUP BY
    client, rank_grouping )
USING (client, rank_grouping)
GROUP BY
  client,
  is_root_page,
  rank_grouping,
  total_in_rank
ORDER BY
  client,
  is_root_page,
  rank_grouping,
  total_in_rank
```

Reply:
I've updated the code to reflect this, but it is difficut to test in the `httparchive.sample_data.pages_1k` as I get, "There is no data to display." when I run it.

Reply:
You need to comment out the date for sample_data as it won’t be for the June dataset.

Reply:
Right.. That works fine now..  Thanks!